osu.Game/Database/ImportTask.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.IO;
+using osu.Game.IO.Archives;
+using osu.Game.Utils;
+using SharpCompress.Common;
+
+namespace osu.Game.Database
+{
+    /// <summary>
+    /// An encapsulated import task to be imported to an <see cref="ArchiveModelManager{TModel,TFileModel}"/>.
+    /// </summary>
+    public class ImportTask
+    {
+        /// <summary>
+        /// The path to the file (or filename in the case a stream is provided).
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// An optional stream which provides the file content.
+        /// </summary>
+        public Stream Stream { get; }
```

Comment:
Maybe worth chucking a `[CanBeNull]` onto this or even doing `#nullable enable` and making this a `Stream?` for additional compiler-level safety?

I would do this myself but not 100% sure what the preferred method would be.